### PR TITLE
[GDB-9925] Module restructuring

### PR DESCRIPTION
**Pull request by simeonzhekofff**
## Description

Module restructuring in order to be more closer to the structure of the Azure Terraform Module.
Also added toggle for the monitoring to have an option to enable it or disable it.
Added definitions for the was provider in order to split region-based resources related to the monitoring.

## Changes
1. Module restructuring
2. Monitoring toggle
3. AWS Provider definition changes
4. Removed unused variables.

## Checklist

- [X] I have tested these changes thoroughly.
- [X] My code follows the project's coding style.
- [X] I have added appropriate comments to my code, especially in complex areas.
- [X] All new and existing tests passed locally.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -139,3 +67,70 @@ module "monitoring" {
   web_test_availability_request_url = module.load_balancer.lb_dns_name
   measure_latency                   = var.monitoring_route53_measure_latency
 }
+
+module "graphdb" {
+  source               = "./modules/graphdb"
```
Comment (viktor-ribchev) [inline main.tf:72]: Please use consistent ordering of the properties along the modules. 
e.g., 

```
module "alabala" {
  source = ""
  <empty line>
  count = ""
  <empty line>
  aws_region = ""
  ..
  ..
}
```

File: modules/graphdb/iam.tf [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    actions = [
+      "ssm:DescribeParameters"
+    ]
+
+    resources = ["arn:aws:ssm:${var.aws_region}:${var.aws_subscription_id}:*"]
+  }
+}
+
+data "aws_iam_policy_document" "instance_volume" {
```
Comment (viktor-ribchev) [inline modules/graphdb/iam.tf:77]: Please use consistent resource naming either `_` or `-`

File: modules/graphdb/variables.tf [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+variable "graphdb_admin_password" {
+  description = "Password for the 'admin' user in GraphDB."
+  type        = string
+  sensitive   = true
+}
+
+variable "graphdb_cluster_token" {
+  description = "Cluster token used for authenticating the communication between the nodes."
+  sensitive   = true
```
Comment (viktor-ribchev) [inline modules/graphdb/variables.tf:39]: no type

File: modules/graphdb/variables.tf [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+variable "instance_type" {
+  description = "EC2 instance type"
+  type        = string
+}
+
+variable "aws_region" {
+  description = "AWS region where GraphDB is being deployed"
+  type        = string
```
Comment (viktor-ribchev) [inline modules/graphdb/variables.tf:21]: no type

File: modules/vpc/variables.tf [hunk 5]
```diff
@@ -1,8 +1,3 @@
-variable "azs" {
```
Comment (viktor-ribchev) [inline modules/vpc/variables.tf:1]: Should remove it from variables.tf as well.

File: modules/graphdb/variables.tf [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+variable "node_count" {
+  description = "Number of GraphDB nodes to deploy in ASG"
+  type        = number
+  default     = 3
+
+  validation {
+    condition     = var.node_count % 2 == 1 && var.node_count > 1
+    error_message = "Node count must be an odd number. Suggested to be 3, 5 or 7"
```
Comment (viktor-ribchev) [inline modules/graphdb/variables.tf:227]: The userdata script does not support anything but 3 nodes at the moment.

File: modules/vpc/variables.tf (hunk 5 above)
Comment (simeonzhekofff) [reply modules/vpc/variables.tf:1]: Removed.

File: modules/graphdb/variables.tf (hunk 4 above)
Comment (simeonzhekofff) [reply modules/graphdb/variables.tf:21]: Fixed

File: modules/graphdb/variables.tf (hunk 3 above)
Comment (simeonzhekofff) [reply modules/graphdb/variables.tf:39]: Fixed.

File: modules/graphdb/iam.tf (hunk 2 above)
Comment (simeonzhekofff) [reply modules/graphdb/iam.tf:77]: Fixed.

File: main.tf (hunk 1 above)
Comment (simeonzhekofff) [reply main.tf:72]: Fixed.

**Review by viktor-ribchev — APPROVED** [review]

**Review by yaskoo — APPROVED** [review]

File: modules/graphdb/variables.tf [hunk 7]
```diff
@@ -0,0 +1,230 @@
+
```
Comment (mihailradkov) [inline modules/graphdb/variables.tf:1]: Resolve the issues from `tflint`

**Review by mihailradkov — APPROVED** [review]

**Pull request merged**